.changeset/layered-env-files.md

```
@@ -0,0 +1,6 @@
+---
+'@mastra/deployer': patch
+'mastra': patch
+---
+
+Layer default dotenv files from base to environment-specific overrides and preserve inherited shell environment variables in `mastra dev`.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)